drivers/amazonec2/amazonec2.go

```diff
@@ -1109,37 +1114,35 @@ func (d *Driver) securityGroupAvailableFunc(id string) func() bool {
 	}
 }
 
-func (d *Driver) configureTags(tagGroups string) error {
-
-	tags := []*ec2.Tag{}
-	tags = append(tags, &ec2.Tag{
-		Key:   aws.String("Name"),
-		Value: &d.MachineName,
-	})
-
-	if tagGroups != "" {
-		t := strings.Split(tagGroups, ",")
-		if len(t) > 0 && len(t)%2 != 0 {
-			log.Warnf("Tags are not key value in pairs. %d elements found", len(t))
-		}
-		for i := 0; i < len(t)-1; i += 2 {
-			tags = append(tags, &ec2.Tag{
-				Key:   &t[i],
-				Value: &t[i+1],
+// buildResourceTags accepts a list of AWS resources that should be tagged
+// upon creation of the EC2 instance. Driver.Tags will be applied to all resources
+// supplied, except for the ec2InstanceResource which will also have the
+// MachineName added as a tag.
+//
+// NB: The ec2InstanceResource must be passed for the EC2 instance to have a name.
+func (d *Driver) buildResourceTags(resources []string) []*ec2.TagSpecification {
+	tags := buildEC2Tags(d.Tags)
+	var tagSpecs []*ec2.TagSpecification
```

Comment:
If you know what the length of a slice will be, then you should set the capacity on initialization to avoid extra allocations.
```suggestion
	tagSpecs := make([]*ec2.TagSpecification, 0, len(resources) + 1)
```